docs/configuration/index.md

```diff
@@ -890,9 +890,9 @@ These Coordinator static configurations can be defined in the `coordinator/runti
-|`druid.coordinator.kill.on`|Boolean flag for whether or not the Coordinator should submit kill task for unused segments, that is, permanently delete them from metadata store and deep storage. If set to true, then for all whitelisted datasources (or optionally all), Coordinator will submit tasks periodically based on `period` specified. A whitelist can be set via dynamic configuration `killDataSourceWhitelist` described later.<br /><br />When `druid.coordinator.kill.on` is true, segments are eligible for permanent deletion once their data intervals are older than `druid.coordinator.kill.durationToRetain` relative to the current time. If a segment's data interval is older than this threshold at the time it is marked unused, it is eligible for permanent deletion immediately after being marked unused.|false|
+|`druid.coordinator.kill.on`|Boolean flag to enable the Coordinator to submit a kill task for unused segments and delete them permanently from the metadata store and deep storage.|false|
 |`druid.coordinator.kill.period`| The frequency of sending kill tasks to the indexing service. The value must be greater than or equal to `druid.coordinator.period.indexingPeriod`. Only applies if kill is turned on.|Same as `druid.coordinator.period.indexingPeriod`|
-|`druid.coordinator.kill.durationToRetain`|Only applies if you set `druid.coordinator.kill.on` to `true`. This value is ignored if `druid.coordinator.kill.ignoreDurationToRetain` is `true`. Valid configurations must be a ISO8601 period. Druid will not kill unused segments whose interval end date is beyond `now - durationToRetain`. `durationToRetain` can be a negative ISO8601 period, which would result in `now - durationToRetain` to be in the future.<br /><br />Note that the `durationToRetain` parameter applies to the segment interval, not the time that the segment was last marked unused. For example, if `durationToRetain` is set to `P90D`, then a segment for a time chunk 90 days in the past is eligible for permanent deletion immediately after being marked unused.|`P90D`|
+|`druid.coordinator.kill.durationToRetain`|Period to retain unused segments in ISO 8601 duration format. When `druid.coordinator.kill.on` is true, segments become eligible for permanent deletion once their data intervals become older than `durationToRetain` relative to the current time. For example, if `durationToRetain` is set to `P90D`, a segment from a time interval 90 days in the past becomes eligible for permanent deletion immediately after being marked unused. If you set `durationToRetain` to a negative ISO 8601 period, `now - durationToRetain` falls in the future.|`P90D`|
```

Comment:
Unfortunately, the config names are a little confusing.
The `bufferPeriod` should really be called the `durationToRetain` since the buffer period represents the duration for which we retain the segments even after they have been marked as unused.
The `durationToRetain` however is the eligibility criteria for segments to be deleted.
I have tried to update the language below to reflect that.

```suggestion
|`druid.coordinator.kill.durationToRetain`|Duration (in ISO 8601 format) relative to the current time identifying data interval of segments which should be retained. When `druid.coordinator.kill.on` is true, any segment with a data interval ending before `now - durationToRetain` is eligible for permanent deletion. For example, if `durationToRetain` is set to `P90D`, an unused segment with a time interval ending 90 days in the past is eligible for deletion. If `durationToRetain` is set to a negative ISO 8601 period, segments with future intervals ending before `now - durationToRetain` would be eligible for deletion too.|`P90D`|
```

Reply:
Updated